pkg/api/controller_test.go

```diff
@@ -1677,46 +1740,56 @@ func TestHTTPReadOnly(t *testing.T) {
 				}(c)
 				// wait till ready
 				for {
-					_, err := resty.R().Get(BaseURL4)
+					_, err := resty.R().Get(baseURL)
 					if err == nil {
 						break
 					}
 					time.Sleep(100 * time.Millisecond)
 				}
-				defer func(controller *api.Controller) {
-					ctx := context.Background()
-					_ = controller.Server.Shutdown(ctx)
-				}(c)
-				// with creds, should get expected status code
-				resp, _ := resty.R().SetBasicAuth(user, password).Get(BaseURL4 + "/v2/")
-				So(resp, ShouldNotBeNil)
-				So(resp.StatusCode(), ShouldEqual, 200)
 
 				// with creds, should get expected status code
-				resp, _ = resty.R().SetBasicAuth(user, password).Get(BaseURL4 + "/v2/")
+				resp, _ := resty.R().SetBasicAuth(user, password).Get(baseURL + "/v2/")
+				t.Log("Response Info:")
+				t.Logf("  Error      : %v", err)
+				t.Logf("  Status Code: %v", resp.StatusCode())
+				t.Logf("  Status     : %v", resp.Status())
+				t.Logf("  Time       : %v", resp.Time())
+				t.Logf("  Received At: %v", resp.ReceivedAt())
+				t.Logf("  Body       : %v \n", resp)
+
+				t.Log(resp)
+
 				So(resp, ShouldNotBeNil)
 				So(resp.StatusCode(), ShouldEqual, 200)
 
 				// with creds, any modifications should still fail on read-only mode
 				resp, err = resty.R().SetBasicAuth(user, password).
-					Post(BaseURL4 + "/v2/" + AuthorizedNamespace + "/blobs/uploads/")
+					Post(baseURL + "/v2/" + AuthorizedNamespace + "/blobs/uploads/")
 				So(err, ShouldBeNil)
 				So(resp, ShouldNotBeNil)
 				So(resp.StatusCode(), ShouldEqual, 405)
 
 				//with invalid creds, it should fail
-				resp, _ = resty.R().SetBasicAuth("chuck", "chuck").Get(BaseURL4 + "/v2/")
+				resp, _ = resty.R().SetBasicAuth("chuck", "chuck").Get(baseURL + "/v2/")
 				So(resp, ShouldNotBeNil)
 				So(resp.StatusCode(), ShouldEqual, 401)
+
+				defer func(controller *api.Controller) {
```

Comment:
Is this change still needed after the update to use 'freeport'?

Reply:
Yes, it works now. I have update it.